sdk/lib/opentelemetry/sdk/configurator.rb

```diff
@@ -210,12 +211,16 @@ def fetch_propagator(name, class_name, gem_suffix = name)
         nil
       end
 
-      def fetch_exporter(name, class_name)
-        Trace::Export::BatchSpanProcessor.new(Kernel.const_get(class_name).new)
+      def fetch_exporter(name, class_name, exporter_options = {})
+        Trace::Export::BatchSpanProcessor.new(Kernel.const_get(class_name).new(**exporter_options), metrics_reporter: metrics_reporter)
```

Comment:
If we aren't documenting this feature, i think it's fine to leave as is. Right now i think the goal is, give power users the ability to insert their own metric reporter provider. If we plan to document or make this something a little more user facing, it might be useful to validate that the methods and method signatures for `add_to_counter`, `record_value`, and `observe_value` exist and are correct, since it'll end up creating runtime errors if not. I don't think this is blocking just thinking out loud about ways this could go wrong in the future.

Reply:
i think thats a good idea, if it becomes a pain then we can drop it

Reply:
I have added a type check. Reporter implementations will have to extend the module which will provide a minimal set of no-op functions. 

```console
irb(main):001:0> require 'opentelemetry-sdk'
=> true
irb(main):002:1* class Reporter
irb(main):004:1*   def add_to_counter(metric, increment: 1, labels: {}); puts "got #{metric} #{increment} #{labels}" ; end
irb(main):005:0> end
=> :add_to_counter
irb(main):006:0> Reporter.new.add_to_counter("hello", labels: { foo: :bar })
got hello 1 {:foo=>:bar}
irb(main):008:0> Reporter.new.record_value("hello", value: 10, labels: { foo: :bar })
=> nil
```